mne/time_frequency/tfr.py

```diff
@@ -1065,7 +1065,7 @@ def save(self, fname, overwrite=False):
 def _prepare_write_tfr(tfr, condition):
     """Aux function"""
     return (condition, dict(times=tfr.times, freqs=tfr.freqs,
-                            data=tfr.data, info=tfr.info, nave=tfr.nave,
+                            data=tfr.data, info=tfr.info._store, nave=tfr.nave,
```

Comment:
I was wrong. Doing a `dict(info)` doesn't solve the H5IO issue, since `'ch_names'` is still a `_ChannelNameList` and not a plain list.

Two ways out of this:

1. Use the original version of storing `info._store`. This works, but doesn't write the `'ch_names'` and `'nchan'` fields, which means the resulting hdf5 file is a little bit different.
2. To retain full backwards compatibility, add an `Info.to_dict` method that also casts `'ch_names'` to a plain list.

Right now, I've implemented the second option, since I'm striving for optimal backwards compatibility with this PR. But other suggestions are appreciated.